### PR TITLE
Upgraded atlas and om go clients

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by threebee — DISMISSED** [review]
LGTM!

File: internal/store/store.go
Comment (gssbzn) [inline internal/store/store.go:180]: 💯

**Comment by andreaangiolillo** [comment]
e2e tests are failing, I am looking into it

File: internal/store/store.go
Comment (gssbzn) [inline internal/store/store.go:274]: I think now `PublicAuthenticatedPreset` and `PrivateAuthenticatedPreset` should work the same right? should we go to just one?

Comment (gssbzn) [reply internal/store/store.go:274]: same for `PublicUnauthenticatedPreset` and `PrivateUnauthenticatedPreset`

**Review by gssbzn — APPROVED** [review]
LGTM keep an eye on the waterfall after merging

Great work with all this

**Pull request merged**
